server/src/services/listings.ts

```diff
@@ -14,14 +14,20 @@
  * limitations under the License.
  */
 
-import {ListingResponse} from 'interfaces';
-
+import {DEFAULT_LISTING_PAYLOAD} from '../constants/default-payload';
+import {ListingRequest, ListingResponse} from '../interfaces';
 import {listingStorage} from '../storage';
 
+const addListing = async (listing: ListingRequest): Promise<ListingResponse> =>
+  listingStorage.addListing({
+    ...listing,
+    ...DEFAULT_LISTING_PAYLOAD,
```

Comment:
I think just to be safe, you should swap the order of these 2 so that listing fields don't get overwritten by the default fields. (even though for this case we know that the `listing` won't contain the fields from the `DEFAULT_LISTING_PAYLOAD`)
```suggestion
    ...DEFAULT_LISTING_PAYLOAD,
    ...listing,
```

Reply:
@hzxcaryn Ohh, thanks, I didn't know that! But, in this case, I actually want the listing fields to be overwritten by default fields because the fields like `numCommits` must be 0 at the point of creation, and similarly with the other fields in `DEFAULT_LISTING_PAYLOAD`.

Reply:
@karenfrilya97 Oh yea, that makes sense for your use case of the default values! haha